src/common/components/select-repository-list/index.js

```diff
@@ -25,18 +30,13 @@ const SNAP_NAME_NOT_REGISTERED_ERROR_CODE = 'snap-name-not-registered';
 export class SelectRepositoryListComponent extends Component {
 
   componentDidMount() {
-    this.props.dispatch(createSnapsClear());
-    this.props.dispatch(unselectAllRepositories());
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const repositoriesStatus = nextProps.repositoriesStatus;
-    const ids = Object.keys(repositoriesStatus);
-    if (ids.length && ids.every((id) => repositoriesStatus[id].success)) {
-      this.props.router.push('/dashboard');
```

Comment:
This could be addressed with some kind of state loading message, because we know the user selected some repos, we know that was successful, we just don't know the result yet.